sherpa/astro/io/__init__.py

```diff
@@ -610,10 +610,24 @@ def read_pha(arg, use_errors=False, use_background=False):
 def _pack_table(dataset):
     """Identify the columns in the data.
 
+    This relies on the _fields attribute containing the
+    data columns, and _extra_fields the extra information
+    (other than the name column).
+
     """
     names = dataset._fields
-    data = {name: getattr(dataset, name) for name in names}
-    return data, names
+    data = {}
+    for name in dataset._fields:
+        if name == 'name':
+            continue
+
+        val = getattr(dataset, name)
+        if val is  None:
+            continue
+
+        data[name] = val
+
+    return data, list(data.keys())
```

Comment:
This is actually a subtle change from the previous version, which said something like

    return data, fields

so I should have pointed out this more. As you said, the caller can just get the keys so it's not needed, but I wasn't thinking about cleaning it up. But as the tests pass (and there's no need that I can see for the old behavior) I'll look at cleaning up this routine.

Reply:
Actually, there is technically a reason for returning the names separate from the dictionary because we would use this to determine the order of the output file. However, we can rely on the >= python 3.7 fact that dictionaries are now ordered,

Reply:
I've simplified the return value from `_pack_table` (although it pushes the logic to the caller).